src/main/java/io/confluent/idesidecar/restapi/models/ConnectionSpec.java

```diff
@@ -1,46 +1,218 @@
 package io.confluent.idesidecar.restapi.models;
 
+import static io.confluent.idesidecar.restapi.models.ConnectionSpec.ConnectionType.CCLOUD;
+import static io.confluent.idesidecar.restapi.models.ConnectionSpec.ConnectionType.DIRECT;
+import static io.confluent.idesidecar.restapi.models.ConnectionSpec.ConnectionType.LOCAL;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import io.confluent.idesidecar.restapi.exceptions.Failure.Error;
+import io.confluent.idesidecar.restapi.util.CCloud.KafkaEndpoint;
+import io.confluent.idesidecar.restapi.util.CCloud.SchemaRegistryEndpoint;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
+import java.util.Optional;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Null;
```

Comment:
I just realized this, but we should be using the Jakarta flavor APIs here instead of Javax. The namespace is compatible, so just replacing `javax` -> `jakarta` should work.

Here's why (courtesy to our internal knowledge base):
The primary reason to transition to Jakarta APIs from javax is due to the transfer of Java EE to the Eclipse Foundation, where it became Jakarta EE.  The older javax packages within Java EE have been deprecated.  Jakarta EE uses the jakarta namespace, which is incompatible with the legacy javax namespace.  Mixing the two can cause issues.  Several projects and frameworks, including Jetty 12, Spring 6, Spring Boot 3, Apache Tomcat 11, and ActiveMQ 6, have already adopted Jakarta.  Additionally, upgrading to Jakarta allows you to keep up with the latest changes and modernize your applications.  In the context of Apache Kafka, upgrading to Jakarta in version 4.0 was considered a good time to make the change as it required incompatible package changes, aligning with the major version release.

Reply:
Further reading (posting for myself to go read later):

- https://jakarta.ee/blogs/javax-jakartaee-namespace-ecosystem-progress/
- https://blog.ltgt.net/javax-jakarta-mess-and-gradle-solution/, and its sequel, https://blog.ltgt.net/javax-jakarta-mess-even-worse/

Reply:
Thanks for pointing that out. I've fixed this PR to use `jakarta.validation.*` instead of `javax.validation`, but we have a few other instances of `javax` imports. I'd prefer to make those changes in a separate PR.

https://github.com/jakartaee/platform/blob/main/namespace/mappings.adoc#mapping-javax-to-jakarta

Reply:
That's fine and yeah I noticed those too. I believe javax is a transitive dependency so until those dependencies get rid of it, we'll keep running into this. I guess a custom checkstyle rule can prevent us from running into it again.